File: 2022/day1/solution.py
#!/usr/bin/python3
all_elves = []
current = 0
with open("input", "r") as filehandle:
    for line in filehandle:
        if line == "\n":
            all_elves.append(current)
            current = 0
        else:
            current += int(line)
all_elves.append(current)
all_elves.sort(reverse=True)
print(f"Part 1: {all_elves[0]}")
print(f"Part 2: {all_elves[0] + all_elves[1] + all_elves[2]}")

File: 2022/day1/solution.pl
#!/usr/bin/perl
use strict;
use warnings;
use feature "say";

my $curr_cals = 0;
my @arr = ();
open my $fh, "<", "input" or die "Cannot open file";
while (<$fh>) {
    chomp $_;
    if ($_ eq "") {
        push @arr, $curr_cals;
        $curr_cals = 0;
    }
    else {
        $curr_cals += $_;
    }
}
push @arr, $curr_cals;
@arr = sort { $b <=> $a } @arr;
say "Part 1: ", $arr[0];
say "Part 2: ", $arr[0] + $arr[1] + $arr[2];
close $fh;

